### ISSUE TITLE
Result dismatch between onnx and image_demo.py

### ISSUE THREAD

**Issue by taofuyu** [AILab-CVC/YOLO-World#169]
I use this [config](https://github.com/AILab-CVC/YOLO-World/blob/3264b61a03b073852b1559fa896cb12c6ff1aa41/configs/pretrain/yolo_world_v2_s_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_lvis_minival.py) and its weight to run image_demo.py  by  --text 'person,tree', the result is:
![000000000872](https://github.com/AILab-CVC/YOLO-World/assets/13781923/c7bc1441-0f49-4ce0-adaa-a8cc5bf0d7dc)

I export this pre-trained model to onnx, and then inference this onnx model, the result is:
![dd](https://github.com/AILab-CVC/YOLO-World/assets/13781923/0dc98da1-7451-43a9-b222-aa3f8a93030e)

As you can see, these scores are different. I have checked that input images are same.

**Comment by wondervictor** [comment]
@taofuyu, are the categories the same between the two models?

**Comment by taofuyu** [comment]
> @taofuyu, are the categories the same between the two models?

yes, when exporting ONNX, I use the same categories as image_demo.py. You could verify this, maybe I mistake something.

**Comment by taofuyu** [comment]
[infer_onnx.txt](https://github.com/AILab-CVC/YOLO-World/files/14715980/infer_onnx.txt)
 And this is my script to inference onnx, maybe can help someone. It detaches `YOLOv5KeepRatioResize` and `LetterResize` from mmyolo, which are used in the test_pipeline of config, to ensure the input of onnx is same as image_demo.py.

**Comment by wondervictor** [comment]
Nice work, I'll check it.

**Comment by taofuyu** [comment]
updates: a very strange phenomenon, parameters in exported onnx and corresponding checkpoint are different. Take the BN of the first conv layer as an example:
![1711097302143](https://github.com/AILab-CVC/YOLO-World/assets/13781923/0618dddb-72c5-472c-baa8-c7f35db5e51c)
![1711097351440](https://github.com/AILab-CVC/YOLO-World/assets/13781923/c58e27b6-ce93-4d23-b350-6d9e35510af9)

**Comment by wondervictor** [comment]
BN and Convolution are fused. See: https://pytorch.org/tutorials/intermediate/custom_function_conv_bn_tutorial.html

**Comment by taofuyu** [comment]
Hello, any updates ?

**Comment by wondervictor** [comment]
Hi @taofuyu, I'll check it as soon as possible in these days. Please let me know if you have any updates.

**Comment by ccl-private** [comment]
Hi, how's it going so far? It's been almost 3 months.

**Comment by ccl-private** [comment]
I found this issue on mmdeploy. https://github.com/open-mmlab/mmdeploy/issues/1903
I don't know if there is any direct connection between the two

**Comment by ccl-private** [comment]
@wondervictor @taofuyu
